rskj-core/src/main/java/co/rsk/net/light/LightProcessor.java

```diff
@@ -0,0 +1,222 @@
+     * @param requestId the id of the request
+     * @param blockHash   the requested block hash.
+     * @param lightPeer
+     */
+    public void processGetBlockReceiptsMessage(long requestId, byte[] blockHash, LightPeer lightPeer) {
+        String blockHashLog = Hex.toHexString(blockHash);
+        logger.trace("Processing block receipts request {} block {}", requestId, blockHashLog);
+        final Block block = blockStore.getBlockByHash(blockHash);
+
+        if (block == null) {
+            // Don't waste time sending an empty response.
+            return;
+        }
+
+        List<TransactionReceipt> receipts = new LinkedList<>();
+
+        for (Transaction tx :  block.getTransactionsList()) {
+            TransactionInfo txInfo = blockchain.getTransactionInfo(tx.getHash().getBytes());
+            receipts.add(txInfo.getReceipt());
+        }
+
+        Message responseMessage = new BlockReceiptsMessage(requestId, receipts);
+        lightPeer.sendMessage(responseMessage);
+    }
+
+    public void processBlockReceiptsMessage(long id, List<TransactionReceipt> blockReceipts, LightPeer lightPeer) {
+        throw new UnsupportedOperationException("Not supported BlockReceipt processing");
+    }
+
+    public void processGetTransactionIndex(long id, byte[] hash, LightPeer lightPeer) {
+        logger.debug("Get Transaction Index Message Received");
+
+        TransactionInfo txinfo = blockchain.getTransactionInfo(hash);
+
+        if (txinfo == null) {
+            // Don't waste time sending an empty response.
+            return;
+        }
+
+        byte[] blockHash = txinfo.getBlockHash();
+        long blockNumber = blockchain.getBlockByHash(blockHash).getNumber();
+        long txIndex = txinfo.getIndex();
+
+        TransactionIndexMessage response = new TransactionIndexMessage(id, blockNumber, blockHash, txIndex);
+        lightPeer.sendMessage(response);
+    }
+
+    public void processTransactionIndexMessage(long id, long blockNumber, byte[] blockHash, long txIndex, LightPeer lightPeer) {
+        throw new UnsupportedOperationException("Not supported TransactionIndexMessage processing");
+    }
+
+    public void processGetCodeMessage(long requestId, byte[] blockHash, byte[] address, LightPeer lightPeer) {
+        String blockHashLog = Hex.toHexString(blockHash);
+        String addressLog = Hex.toHexString(address);
+        logger.trace("Processing code request {} block {} code {}", requestId, blockHashLog, addressLog);
+
+        final Block block = blockStore.getBlockByHash(blockHash);
+
+        if (block == null) {
+            // Don't waste time sending an empty response.
+            return;
+        }
+
+        RepositorySnapshot repositorySnapshot = repositoryLocator.snapshotAt(block.getHeader());
+        RskAddress addr = new RskAddress(address);
+        Keccak256 codeHash = repositorySnapshot.getCodeHash(addr);
+
+        CodeMessage response = new CodeMessage(requestId, codeHash.getBytes());
```

Comment:
If this is modelled after https://openethereum.github.io/wiki/The-Parity-Light-Protocol-%28PIP%29 then shouldn't this be the bytecode rather than the code hash?

Reply:
@smcsicardi

Reply:
Totally right! Clear mistake on my part, will change it now.